table/tables/tables.go

```diff
@@ -326,6 +326,22 @@ func adjustRowValuesBuf(sessVars *variable.SessionVars, rowLen int) {
 	sessVars.AddRowValues = sessVars.AddRowValues[:adjustLen]
 }
 
+// getRollbackableMemStore get a rollbackable BufferStore, when we are importing data,
+// Just add the kv to transaction's membuf directly. the returned *kv.BufferStore avoids
+// type assert in bs.SaveTo(txn).
+func (t *Table) getRollbackableMemStore(ctx context.Context) (kv.RetrieverMutator, *kv.BufferStore) {
+	if ctx.GetSessionVars().ImportingData {
+		return ctx.Txn(), nil
+	}
+
+	bs := ctx.GetSessionVars().BufStore
+	if bs == nil {
+		bs = kv.NewBufferStore(ctx.Txn(), kv.DefaultTxnMembufCap)
```

Comment:
ctx.GetSessionVars().BufStore = bs ?

Reply:
ctx.GetSessionVars().BufStore is new in InsertExec.exec, if it is nil, means we are in other executor and not use sessVars.BufStore yet. I will refactor these buffers code to make it clear and graceful.